vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/vaadin-grid-flow-selection-column.js

```diff
@@ -42,6 +49,12 @@ export class GridFlowSelectionColumn extends GridSelectionColumnBaseMixin(GridCo
     }
   }
 
+  /** @private */
+  __shiftKeyDownChanged(newState, oldState) {
+    if (oldState !== undefined) {
+      this.$server.setShiftKeyDown(newState);
```

Comment:
This causes an extra server roundtrip once the user releases the shift key (after doing a range selection)

Reply:
Yeah, I happened upon this too, but I assumed that it's not a big deal. One optimization I was considering is to pass the shift key state through `$server.select` / `$server.deselect`, but that would consequently force us to change the signature of public  `selectFromClient` and `deselectFromClient` methods to accept the second parameter:

https://github.com/vaadin/flow-components/blob/bc3fc8b2eef4143bae3db1063c4826b1ea70b3ac/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/GridSelectionModel.java#L34

Another idea I had was to set a flag before calling `selectFromClient`. However, I wasn't sure enough if it's a safe approach:

```diff
  @ClientCallable
- private void select(String key) {
+ private void select(String key, boolean isShiftKeyDown) {
+     getSelectionModel().setShiftKeyDown(isShiftKeyDown);
      findByKey(String.valueOf(key))
              .ifPresent(getSelectionModel()::selectFromClient);
+     getSelectionModel().setShiftKeyDown(false);
  }
```

Reply:
> One way to fix it could be to dispatch an event for this instead and use addEventListener(...).debounce(...) on the server.

I'm not sure I get this approach. What should be the debounce condition?

Reply:
> I'm not sure I get your suggestion. What should be the debounce condition?

I was thinking of something like
```java
getElement()
    .addEventListener("shift-key-toggle", this::shiftKeyToggle)
    .debounce(Integer.MAX_VALUE);
```

Maybe there's some more elegant way

Reply:
After some thinking, it seems unnecessary to reset the shift key state after the selection and it's only necessary to ensure that it's updated on the server before `selectFromClient` is called.